### ISSUE TITLE
[Change Request] deleting a single backglass triggers expensive reload

### ISSUE THREAD

**Issue by gorgatron1** [syd711/vpin-studio#512]
**Describe the bug**

deleting a single backglass from the back glasses tab triggers expensive reload

**To Reproduce**

1. go to Backglasses tab
2. delete a backglass, e.g. an orphaned file
3. most modifications are very quick, but deleting a file seems to invalidate the data and force a full reload
4. on my machine (774 back glasses) this takes ~25 seconds

```
05-10 11:19:06.006 INFO  [ForkJoinPool.commonPool-worker-19] o.s.w.c.RestTemplate - HTTP GET http://pinbot.local:8089/api/v1/directb2s (25166ms)
```

**Expected behavior**

Ideally it would just remove the row and not invalidate, or if it needs to invalidate, perhaps it can be faster.

**Desktop (please complete the following information):**
 - OS: macOS 15

**Comment by gorgatron1** [comment]
Another time I did this it just removed the row without doing a full reload, so not a reliable repro.

**Comment by syd711** [comment]
@leprinco I've saved the backglass fetch result and remove the entry without re-fetching all data.
Nevertheless, we should check if we can provide a better caching here, so I'll leave the ticket open.

**Comment by leprinco** [comment]
fixed by updating the models.
Also removed the global reload() when renaming and duplicating a B2S
Also properly managed the discovery of a new backglass when table is updated

**Comment by syd711** [comment]
Fixed with 3.7.2.
